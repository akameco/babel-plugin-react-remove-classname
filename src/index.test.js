// @flow
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  tests: [
    {
      title: 'remove test-hello',
      code: `
  const hello = () => (
    <div className="test-hello"></div>
  )
`,
    },
    {
      title: 'remove test-other',
      code: `
  const hello = () => (
    <div className="test-other"></div>
  )
`,
    },
    {
      title: 'remove some className',
      code: `
  const hello = () => (
    <div className="test-a hello test-b"></div>
  )
`,
    },
    {
      title: 'no remove',
      code: `
  const hello = () => (
    <div data-test="test-hello"></div>
  )
`,
    },
    {
      title: 'with expresstion',
      code: `
  const hello = ({ isActive }) => (
    <div className={isActive ? "test-hello" : "hello"}>hello</div>
  )
`,
    },
  ],
})

pluginTester({
  title: 'with option',
  plugin,
  snapshot: true,
  pluginOptions: { prefix: 'babel-' },
  tests: [
    {
      title: 'with prefix',
      code: `
  const hello = () => (
    <div className="babel-a hello test-b"></div>
  )
`,
    },
  ],
})
