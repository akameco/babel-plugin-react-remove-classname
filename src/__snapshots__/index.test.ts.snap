// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default no remove: no remove 1`] = `

const hello = () => (
  <div data-test="test-hello"></div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const hello = () => <div data-test="test-hello"></div>;

`;

exports[`default remove some className: remove some className 1`] = `

const hello = () => (
  <div className="test-a hello test-b"></div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const hello = () => <div className="hello"></div>;

`;

exports[`default remove test-hello: remove test-hello 1`] = `

const hello = () => (
  <div className="test-hello"></div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const hello = () => <div className=""></div>;

`;

exports[`default remove test-other: remove test-other 1`] = `

const hello = () => (
  <div className="test-other"></div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const hello = () => <div className=""></div>;

`;

exports[`default with expresstion: with expresstion 1`] = `

const hello = ({ isActive }) => (
  <div className={isActive ? "test-hello" : "hello"}>hello</div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const hello = ({
  isActive
}) => <div className={isActive ? "test-hello" : "hello"}>hello</div>;

`;

exports[`with option with prefix: with prefix 1`] = `

const hello = () => (
  <div className="babel-a hello test-b"></div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const hello = () => <div className="hello test-b"></div>;

`;
